### ISSUE TITLE
Parser failure for imports with .js extension

### ISSUE THREAD

**Issue by orgads** [francescov1/mongoose-tsgen#160]
enums.ts:
```ts
export enum MyEnum {
  VAL1 = 'Val1',
  VAL2 = 'Val2'
}
```

schema.ts:
```ts
import { MyEnum } from './enums.js';
```

Error:
```
Generating mongoose typescript definitions... !
    Error: Cannot find module './enums.js'
    Require stack:
    - /project/schema.ts
    - /project/node_modules/mongoose-tsgen/lib/parser/utils.js
    - /project/node_modules/mongoose-tsgen/lib/parser/schema.js
    - /project/node_modules/mongoose-tsgen/lib/helpers/generator.js
    - /project/node_modules/mongoose-tsgen/lib/index.js
    - /project/node_modules/mongoose-tsgen/bin/run
    Code: MODULE_NOT_FOUND
```

**Comment by francescov1** [comment]
@orgads You're trying to import a TS file as if its a JS file. Changing `import { MyEnum } from './enums.js';` to `import { MyEnum } from './enums.ts';` or `import { MyEnum } from './enums';` should fix it

**Comment by orgads** [comment]
This is how imports are done in esm. This project is cjs, but the imports are esm compatible. 

Changing it to `.ts` won't work.

**Comment by francescov1** [comment]
Enums are only supported in Typescript, so the `enum` file would have to be `enum.ts`. I dont think you can import a ts file using a `js` extension can you?

**Comment by francescov1** [comment]
Got it, thanks for clearing up, havent seen it befpre. Any chance you can provide a simple reproduction repo and ill add support for it?

**Comment by orgads** [comment]
https://github.com/orgads/js-import

**Comment by francescov1** [comment]
Thanks @orgads will look this weekend

**Comment by francescov1** [comment]
Fixed in v9.4.2
